nntrainer/layers/layer_node.h

```diff
@@ -331,7 +331,9 @@ class LayerNode final : public ml::train::Layer, public GraphNode {
    */
   void addInputLayers(const std::string &in_layer) {
     input_layers.push_back(in_layer);
-    layerv1->setNumInputs(input_layers.size());
+    input_dim.resize(input_layers.size());
```

Comment:
So the input_dim is managed in layer_node and also in layer_context?

Reply:
That's a good question.
`input_dim` in `LayerNode` stores the intermediate value of `input_dim` till `finalize` is called. 
The reason to do this way is to avoid an interface in `InitLayerContext` which allows modifying `input_dim` to layer developers.
I am open to ideas for this. Please propose a new design for this, and I will update this PR :).